packages/core/src/dialects/mssql/query-generator-typescript.ts

```diff
@@ -83,4 +84,30 @@ export class MsSqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
       this.quoteTable(tableName),
     ]);
   }
+
+  jsonPathExtractionQuery(sqlExpression: string, path: ReadonlyArray<number | string>, _unquote: boolean): string {
+    let jsonPathStr = '$';
+    for (const pathElement of path) {
+      if (typeof pathElement === 'number') {
+        jsonPathStr += `[${pathElement}]`;
+      } else {
+        jsonPathStr += `.${this.#quoteJsonPathIdentifier(pathElement)}`;
+      }
+    }
```

Comment:
This code + `quoteJsonPathIdentifier` could be moved to a "buildJsonPath" function in our utils to reduce copy/pasting